TiFlash segmentation fault when ~mpptask

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
chbenchmark 1000 warehouse workload
dbaas env
### 2. What did you expect to see? (Required)
no crash
### 3. What did you see instead (Required)
```
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:13. /lib64/libpthread.so.0(+0x7ea5) [0x7f7ee772cea5]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:12. /tiflash/tiflash() [0x8ab5bef]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:11. /tiflash/tiflash(_ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJZN2DB13ThreadFactory9newThreadISt8functionIFvvEEJEEES_bNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEOT_DpOT0_EUlDpOT_E_EEEEE6_M_runEv+0x63) [0x7da8e83]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:10. /tiflash/tiflash() [0x7c6eaf1]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:9. /tiflash/tiflash(std::_Sp_counted_ptr<DB::MPPTask*, (__gnu_cxx::_Lock_policy)2>::_M_dispose()+0x1a) [0x7c687aa]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:8. /tiflash/tiflash(DB::MPPTask::~MPPTask()+0x538) [0x7c6b008]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:7. /tiflash/tiflash() [0x36e2bd2]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:6. /tiflash/tiflash(DB::ProcessListEntry::~ProcessListEntry()+0x24) [0x79d3dc4]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:5. /tiflash/tiflash(DB::ProcessListElement::releaseQueryStreams()+0x1b2) [0x79d3702]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:4. /tiflash/tiflash(DB::UnionBlockInputStream<(DB::StreamUnionMode)0, true>::~UnionBlockInputStream()+0x2b5) [0x7c4a435]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:3. /tiflash/tiflash(DB::IProfilingBlockInputStream::~IProfilingBlockInputStream()+0x20a) [0x3702dca]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:2. /tiflash/tiflash(DB::ExchangeSender::~ExchangeSender()+0x30a) [0x7ce58ba]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:1. /tiflash/tiflash(DB::ExpressionBlockInputStream::~ExpressionBlockInputStream()+0x3a9) [0x7ab4829]"] [thread_id=1160]
[2022/01/18 14:26:14.194 +00:00] [ERROR] [<unknown>] ["BaseDaemon:0. /tiflash/tiflash(std::_Sp_counted_ptr_inplace<DB::ExpressionActions, std::allocator<DB::ExpressionActions>, (__gnu_cxx::_Lock_policy)2>::_M_dispose()+0x110) [0x7900b70]"] [thread_id=1160]
[2022/01/18 14:26:14.073 +00:00] [ERROR] [<unknown>] ["BaseDaemon:Attempted access has violated the permissions assigned to the memory area."] [thread_id=1160]
[2022/01/18 14:26:14.073 +00:00] [ERROR] [<unknown>] ["BaseDaemon:Access: write."] [thread_id=1160]
[2022/01/18 14:26:14.073 +00:00] [ERROR] [<unknown>] ["BaseDaemon:Address: 0xa418578"] [thread_id=1160]
[2022/01/18 14:26:14.073 +00:00] [ERROR] [<unknown>] ["BaseDaemon:(from thread 1157) Received signal Segmentation fault (11)."] [thread_id=1160]
[2022/01/18 14:26:14.073 +00:00] [ERROR] [<unknown>] [BaseDaemon:########################################] [thread_id=1160] 
```
### 4. What is your TiFlash version? (Required)
```
+---------+--------------+------------------------------------------+
| type    | max(version) | max(git_hash)                            |
+---------+--------------+------------------------------------------+
| tikv    | 5.4.0        | 6dec088af4fc22764c22e1758074402b1d8b82b1 |
| tidb    | 5.4.0        | af3ed8246cbb959e56ef514321002ca3085d6253 |
| tiflash | 5.4.0        | 01793a4d65be3174053a507df38d1ccc5c19149b |
| pd      | 5.4.0        | e3807695b6fc524f9cb84402937e56e733cabd64 |
+---------+--------------+------------------------------------------+
```

--- a/dbms/src/Flash/Mpp/MPPTask.cpp
+++ b/dbms/src/Flash/Mpp/MPPTask.cpp
@@ -72,7 +72,7 @@ void MPPTask::finishWrite()
 
 void MPPTask::run()
 {
-    newThreadManager()->scheduleThenDetach(true, "MPPTask", [this] { this->shared_from_this()->runImpl(); });
+    newThreadManager()->scheduleThenDetach(true, "MPPTask", [self = shared_from_this()] { self->runImpl(); });
 }
 
 void MPPTask::registerTunnel(const MPPTaskId & id, MPPTunnelPtr tunnel)

